### PR TITLE
[AMORO-2900] Web terminal supports SQL snippets

### LINKED ISSUE #2900: [Improvement]: Web terminal supports run SQL snippets
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

When I click the "Run" button on AMS Terminal, all contents in the editor will be sent to the backend service to execute.

### How should we improve?

Only selected sql snippets should be sent to backend and return related results to the frontend.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

Close #2900.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by tcodehuber** [comment]
image

**Comment by zhoujinsong** [comment]
Great new feature!

Can you help to review it? @zhangmo8 @minteliuwm

File: amoro-ams/amoro-ams-dashboard/src/views/terminal/index.vue [hunk 1]
```diff
@@ -229,9 +229,15 @@ export default defineComponent({
         showDebug.value = true
         resetResult()
         runStatus.value = 'Running'
+
+        let sqlToExecute = sqlEditorRef.value.getSelection()
+        if (!sqlToExecute) {
+          sqlToExecute = sqlSource.value
+        }
```
Comment (zhangmo8) [inline amoro-ams/amoro-ams-dashboard/src/views/terminal/index.vue:236]:
```suggestion
        const sqlToExecute = sqlEditorRef.value.getSelection() || sqlSource.value
```

**Review by zhangmo8 — APPROVED** [review]
LGTM

**Review by zhoujinsong — APPROVED** [review]
@zhangmo8 Thanks for the review!

**Pull request merged**